src/query/api/v1/handler/prometheus/native/complete_tags.go

```diff
@@ -77,15 +80,49 @@ func (h *CompleteTagsHandler) ServeHTTP(w http.ResponseWriter, r *http.Request)
 		return
 	}
 
-	result, err := h.storage.CompleteTags(ctx, query, opts)
-	if err != nil {
+	var (
+		mu       sync.Mutex
+		wg       sync.WaitGroup
+		multiErr xerrors.MultiError
+
+		nameOnly      = tagCompletionQueries.NameOnly
+		resultBuilder = storage.NewCompleteTagsResultBuilder(nameOnly)
+		meta          = block.NewResultMetadata()
+	)
+
+	for _, query := range tagCompletionQueries.Queries {
+		wg.Add(1)
+		// Capture variables.
+		query := query
+		go func() {
+			result, err := h.storage.CompleteTags(ctx, query, opts)
+			mu.Lock()
+			if err != nil {
+				multiErr = multiErr.Add(err)
+				return
```

Comment:
Might also need to do `defer func() { mu.Unlock() ; wg.Done() }()` after the `mu.Lock()` otherwise now the early return stops those running?